### PR TITLE
Added two other required columns and added indexs

### PR DESCRIPTION
**Pull request by ub0005**

### CODE REVIEW

**Comment by NISH1001** [comment]
@ub0005 Looks good. But one thing:
Have we also updated the PATCH request for storage ids? (Just to make sure we have CRUD also...)

**Comment by ub0005** [comment]
> @ub0005 Looks good. But one thing: Have we also updated the PATCH request for storage ids? (Just to make sure we have CRUD also...)

Yes, @NISH1001 Patch requests take care of these additional fields automatically, no code modification is required, as we changed the models and schemas.

**Review by NISH1001 — COMMENTED** [review]
Overall looks good....

File: src/app.py [hunk 1]
```diff
@@ -338,13 +338,19 @@ def upload_csv():
         )
 
     # make sure these columns aren't empty
-    if data[["uuid", "name", "sealed_state"]].isna().sum().sum() > 0:
-        logger.error("uuid or name or sealed_state column values are empty!")
+    if data[CONSTANTS.CATALOGUE_POST_MANDATORY_FIELDS].isna().sum().sum() > 0:
+        logger.error(
+            "Any of the column "
+            + ",".join(CONSTANTS.CATALOGUE_POST_MANDATORY_FIELDS)
+            + " values are empty!"
```
Comment (xhagrg) [inline src/app.py:345]: Reuse variable.

File: src/services/db/models.py [hunk 2]
```diff
@@ -32,18 +32,23 @@ class CatalogueItem(db.Model):
...
+    )
     transfer_checksum_value = db.Column(db.String, nullable=True)
     transfer_checksum_verification = db.Column(db.String(20), nullable=True)
     transfer_started_on = db.Column(db.DateTime, nullable=True)
     transfer_completed_on = db.Column(db.DateTime, nullable=True)
     transfer_source = db.Column(db.String, nullable=True)
     transfer_destination = db.Column(db.String, nullable=True)
 
-    sealed_state = db.Column(db.String)
+    sealed_state = db.Column(db.String, index=True)
```
Comment (xhagrg) [inline src/services/db/models.py:45]: What's a sealed_state?

Comment (ub0005) [reply src/services/db/models.py:45]: The sealed state status indicates the file storage location. There are three sealed states. A sealed state indicates the file is in the cold storage/archived state. An unsealed state means that the file is in the hot storage state and readily available to be transferred. Unsealing state means that the file is being transitioned from cold storage to hot storage.

File: src/app.py (hunk 1 above)
Comment (ub0005) [reply src/app.py]: Reused Error Message

**Comment by xhagrg** [comment]
LGTM.

File: src/constants.py [hunk 3]
```diff
@@ -35,3 +35,9 @@
 }
 
 DATETIME_OLDEST = datetime.datetime(year=1970, month=1, day=1)
+
+ERROR_MSG_ANY_OF_THE_CATALOGUE_POST_MANDATORY_FIELDS_EMPTY = (
```
Comment (xhagrg) [inline src/constants.py:39]: This can be just a simple variable in the app.py file rather than a config.py file entry.

Comment (ub0005) [reply src/constants.py]: moved the variable to the app.py file

File: src/app.py [hunk 4]
```diff
@@ -338,13 +343,15 @@ def upload_csv():
         )
 
     # make sure these columns aren't empty
-    if data[["uuid", "name", "sealed_state"]].isna().sum().sum() > 0:
-        logger.error("uuid or name or sealed_state column values are empty!")
+    if data[CONSTANTS.CATALOGUE_POST_MANDATORY_FIELDS].isna().sum().sum() > 0:
+        logger.error(
+            CONSTANTS.ERROR_MSG_ANY_OF_THE_CATALOGUE_POST_MANDATORY_FIELDS_EMPTY
```
Comment (xhagrg) [inline src/app.py:348]: It's not in the scope of constants now, right?

Comment (ub0005) [reply src/app.py]: yes Iksha, fixed and tested it.

**Comment by xhagrg** [comment]
LGTM.

**Pull request merged**